src/viam/app/app_client.py

```diff
@@ -1581,25 +1632,25 @@ async def delete_robot(self, robot_id: str) -> None:
         request = DeleteRobotRequest(id=robot_id)
         await self._app_client.DeleteRobot(request, metadata=self._metadata)
 
-    async def list_fragments(self, org_id: str, show_public: bool = True) -> List[Fragment]:
+    async def list_fragments(self, org_id: str, show_public: bool = True, visibilities: List[FragmentVisibility] = []) -> List[Fragment]:
```

Comment:
This is a python gotcha where defaulting to an empty list can actually cause bugs (see the *Default parameter values are evaluated from left to right when the function definition is executed.* section of the [python docs](https://docs.python.org/3/reference/compound_stmts.html#function-definitions)). 

So if the actual parameter type should be `visibilities: Optional[List[FragmentVisibility]] = None` and the body of the method should check to see if `visibilities` is `None` and if so, set an empty list.

Reply:
Interesting, thanks for this call out!